src/pages/Admin/Venue/Users/VenueUsersReport.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { useWorldUsers } from "hooks/users";
+import "./VenueUsersReport.scss";
+import Table from "react-bootstrap/Table";
+import Form from "react-bootstrap/Form";
+import { downloadGeneratedCSVFile } from "utils/csv";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+const VenueUsersReport: React.FC = () => {
+  const { worldUsers } = useWorldUsers();
+
+  const [headers, setHeaders] = useState<Array<string>>([]);
+  const [headersSelected, setHeadersSelected] = useState<Array<string>>([]);
+  const [customFieldValue, setCustomFieldValue] = useState("");
+
+  const userIsComplete = useCallback((user: WithId<User>) => {
+    return user.partyName && user.pictureUrl;
+  }, []);
+
+  const getUsersHeaders = useCallback(() => {
+    let index = 0;
+    while (index < worldUsers.length) {
+      if (userIsComplete(worldUsers[index])) {
+        setHeaders(Object.keys(worldUsers[index]));
+        return;
+      } else {
+        index++;
+      }
+    }
+  }, [worldUsers, userIsComplete]);
+
+  useEffect(() => {
+    getUsersHeaders();
+  }, [worldUsers, getUsersHeaders]);
+
+  const renderPageHeader = () => {
+    return <h4>Users Reports</h4>;
+  };
+
+  const renderUserTable = () => {
+    return (
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>Party Name</th>
+            <th>Full Name</th>
+            <th>Department</th>
+          </tr>
+        </thead>
+        <tbody>
+          {/* when I try to use WithId<User> as realName are not added in it, it shows erro, so that's why I need any */}
+          {/* eslint-disable-next-line*/}
+          {worldUsers.map((user: any, index) => {
+            return (
+              userIsComplete(user) && (
+                <tr key={index}>
+                  <td>{index + 1}</td>
+                  <td>{user.partyName || "Not"}</td>
+                  <td>{user.realName || "Not set"}</td>
+                  <td>{user.companyDepartment || "Not set"}</td>
```

Comment:
We use `??` for handling defaults on missing data, not `||`.

I believe we have `const`s in `src/settings` for handling the default `partyName`, etc when not defined.

`realName` and `companyDepartment` don't exist on `staging`, they're an environment specific hack on `env/github`. Once you fix the types as per my suggestion above (never use `any`, and if you think you need to use `any` then the answer is still not to use `any` and to ask someone to help with the types if you need it because there will always be a better solution; That goes in general for disabling `eslint` rules as well. It's SUPER rare that it actually needs to be done.)

```suggestion
                  <td>{user.partyName ?? DEFAULT_PARTY_NAME}</td>
```

Reply:
fixed in 12e5c07fea64a78aced31d706faf59c5e523c733